kratos/python_scripts/vtu_output_process.py

```diff
@@ -0,0 +1,139 @@
+        self.nodal_flags = parameters["nodal_flags"].GetStringArray()
+        self.element_data_value_variables = parameters["element_data_value_variables"].GetStringArray()
+        self.element_flags = parameters["element_flags"].GetStringArray()
+        self.condition_data_value_variables = parameters["condition_data_value_variables"].GetStringArray()
+        self.condition_flags = parameters["condition_flags"].GetStringArray()
+
+        file_format = parameters["file_format"].GetString()
+        if file_format == "ascii":
+            self.writer_format = Kratos.VtuOutput.ASCII
+        elif file_format == "binary":
+            self.writer_format = Kratos.VtuOutput.BINARY
+        else:
+            raise RuntimeError(f"Unsupported file format requested [ requested format = {file_format} ]. Supported file formats:\n\tascii\n\tbinary")
+
+        if parameters["save_output_files_in_folder"].GetBool():
+            self.output_path = Path(parameters["output_path"].GetString())
+            if not self.model_part.ProcessInfo[Kratos.IS_RESTARTED]:
+                kratos_utils.DeleteDirectoryIfExisting(str(self.output_path))
+            self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
+            # now create the output path
+            self.output_path.mkdir(parents=True, exist_ok=True)
+        else:
+            self.output_path = Path(".")
+
+        self.output_interval = parameters["output_interval"].GetDouble()
+        self.output_control = parameters["output_control_type"].GetString()
+        self.next_output = 0.0
+
+        self.vtu_output_ios: 'list[Kratos.VtuOutput]' = []
+
+        self.__ScheduleNextOutput() # required here esp for restart
+
+    def ExecuteInitialize(self) -> None:
+        # check and create all the vtu outputs
+        self.vtu_output_ios.append(Kratos.VtuOutput(self.model_part, not self.write_deformed_configuration, self.writer_format, self.output_precision))
+
+        if self.output_sub_model_parts:
+            for sub_model_part in self.model_part.SubModelParts:
+                self.vtu_output_ios.append(Kratos.VtuOutput(sub_model_part, not self.write_deformed_configuration, self.writer_format, self.output_precision))
+
+        for vtu_output_io in self.vtu_output_ios:
+            self.__AddData(vtu_output_io)
+
+    def PrintOutput(self) -> None:
+        if self.output_control == "time":
+            current_post_fix = str(self.__GetTime())
+        else:
+            current_post_fix = str(self.model_part.ProcessInfo[Kratos.STEP])
+
+        for vtu_output in self.vtu_output_ios:
+            vtu_output.PrintOutput(str(self.output_path / vtu_output.GetModelPart().FullName()) + "_" + current_post_fix)
+
+        self.__ScheduleNextOutput()
+
+    def IsOutputStep(self) -> bool:
+        if self.output_control == "time":
+            return self.__GetTime() >= self.next_output
+        else:
+            return self.model_part.ProcessInfo[Kratos.STEP] >= self.next_output
```

Comment:
membership tests for time/step intervals is a common problem and should be handled in a separate utility.

Reply:
May be in a different PR? I think it makes sense to unify all the outputs (vtk, vtu) using a common mechanism.